camel-core/src/main/java/org/apache/camel/builder/ExpressionBuilder.java

```diff
@@ -1863,9 +1867,30 @@ public String toString() {
         };
     }
 
+    public static Expression dateExpression(final String command) {
+        return dateExpression(command, TimeZone.getDefault().getID(), "");
+    }
+
     public static Expression dateExpression(final String command, final String pattern) {
+        return dateExpression(command, TimeZone.getDefault().getID(), pattern);
+    }
+
+    public static Expression dateExpression(final String commandWithOffsets, final String timezone, final String pattern) {
         return new ExpressionAdapter() {
             public Object evaluate(Exchange exchange) {
+                // Capture optional time offsets
+                String command = commandWithOffsets.split("[+-]", 2)[0].trim();
```

Comment:
Empty command is an error and the method already checks for that later. I don't know if adding a test here would add any value.